### PR TITLE
Add ethereum address validation

### PR DESCRIPTION
**Pull request by james-toussaint**

### CODE REVIEW

**Review by zguesmi — APPROVED** [review]
Nice!

**Review by mcornaton — APPROVED** [review]
Thanks!

File: src/main/java/com/iexec/common/chain/validation/EthereumNonZeroAddressValidator.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
```
Comment (jbern0rd) [inline src/main/java/com/iexec/common/chain/validation/EthereumNonZeroAddressValidator.java]: I don't remember the decision on copyright year.
Do we keep homogeneous values or real year values ?

File: src/main/java/com/iexec/common/chain/validation/EthereumNonZeroAddressValidator.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+package com.iexec.common.chain.validation;
+
+import com.iexec.common.utils.BytesUtils;
+import com.iexec.common.utils.EthAddress;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EthereumNonZeroAddressValidator
```
Comment (jbern0rd) [inline src/main/java/com/iexec/common/chain/validation/EthereumNonZeroAddressValidator.java:25]: Could this be tested with unit tests in order to show valid and invalid use cases ?

File: src/main/java/com/iexec/common/chain/validation/EthereumNonZeroAddressValidator.java (hunk 1 above)
Comment (mcornaton) [reply src/main/java/com/iexec/common/chain/validation/EthereumNonZeroAddressValidator.java]: We strive for real year values :)

Comment (james-toussaint) [reply src/main/java/com/iexec/common/chain/validation/EthereumNonZeroAddressValidator.java]: Fixed copyright thanks!

Copyright dates are tighten to release dates

File: src/main/java/com/iexec/common/chain/validation/EthereumNonZeroAddressValidator.java (hunk 2 above)
Comment (james-toussaint) [reply src/main/java/com/iexec/common/chain/validation/EthereumNonZeroAddressValidator.java:25]: I agree, added test here https://github.com/iExecBlockchainComputing/iexec-common/pull/310/commits/9da2834c0dbc86dc8c83d69d84f3eabf5baef173

**Review by jbern0rd — APPROVED** [review]
Thanks for the tests

**Pull request merged**